### PR TITLE
Igonchar/tcomp 611

### PR DESCRIPTION
**Pull request by gonchar-ivan**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

adds mapping files parser for JDBC components.
It provides similar functionality as Dbms, MappingFileLoader, MappingTypeRetriever classes from
https://github.com/Talend/tcommon-studio-se/tree/master/main/plugins/org.talend.core.runtime/src/main/java/org/talend/core/model/metadata

Note, these classes were refactored. MappingTypeRetriever is not required anymore. All its functionality should be implemented in Dbms class as it was only a wrapper class around data provided by Dbms.
There is also some kind of Manager class available in generated code for component runtime. It was not implemented yet

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://jira.talendforge.org/browse/TCOMP-611

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1320/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1320/)

File: core/components-common/src/main/java/org/talend/components/common/config/jdbc/Dbms.java [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Provides information of type mapping for Database Management System (DBMS)
+ * This data is retrieved from external configuration for JDBC components
+ */
+public class Dbms {
```
Comment (gonchar-ivan) [inline core/components-common/src/main/java/org/talend/components/common/config/jdbc/Dbms.java]: This class is the main "interface" from which you may get required mapping information

File: core/components-common/src/main/java/org/talend/components/common/config/jdbc/MappingFileLoader.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * JDBC configuration parser (mapping_*.xml)
+ */
+public class MappingFileLoader {
```
Comment (gonchar-ivan) [inline core/components-common/src/main/java/org/talend/components/common/config/jdbc/MappingFileLoader.java]: This is mapping files parser. You need specific mapping file location to use this parser. 
This parser returns a list (but there is only one element) of Dbms instances

**Comment by gonchar-ivan** [comment]
One class is missed here, but I'm not sure what features are required from that class.
That class is Manager class from generated code. You may see it:
https://github.com/Talend/tdi-studio-se/blob/79bba735e8bd56d75aa2bc5630b35bf81fbe2cc6/main/plugins/org.talend.designer.components.localprovider/components/templates/db_output_bulk.skeleton#L256-L1217

(It uses MappingTypeRetriever, which uses info from mapping file. So, I think we need to copy required features to components-common)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1341/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1341/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1342/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1342/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1343/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1343/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1345/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1345/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aigonchar%2FTCOMP-611)

File: core/components-common/src/main/java/org/talend/components/common/config/jdbc/AvroTypeConverter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+    /**
+     * Converts Talend type to Avro type schema
+     * 
+     * @param talendType data integration native type
+     * @param logicalType avro logical type
+     * @return field schema
+     * @throws {@link UnsupportedOperationException} in case of unsupported Talend type or logical type
+     */
+    public static Schema convertToAvro(TalendType talendType, String logicalType) {
```
Comment (wwang-talend) [inline core/components-common/src/main/java/org/talend/components/common/config/jdbc/AvroTypeConverter.java]: no talend object type support?

Comment (gonchar-ivan) [reply core/components-common/src/main/java/org/talend/components/common/config/jdbc/AvroTypeConverter.java]: It is also not supported on Studio side - DI Object column type is not converted to some type from Avro during MetadataTable -> Avro schema conversion. We can do a workaround. We can map Object type to Avro String type with JAVA_CLASS_FLAG="Object" property.
If we need, we should add support of it in several places:
here, codegen, MetadataTable -> Avro schema conversion on Studio side. But I have few time for the moment.

Do you need object type support for JDBC feature? If yes, I can start from adding support here.

**Pull request closed without merging**